crates/cheatcodes/src/inspector.rs

```diff
@@ -1400,92 +1431,105 @@ impl Cheatcodes {
         }
 
         if self.use_zk_vm {
-            if let TransactTo::Call(test_contract) = ecx.env.tx.transact_to {
-                if call.bytecode_address == test_contract {
-                    info!("using evm for calls to test contract {:?}", ecx.env);
-                    return None
+            // also skip if the target was created during a zkEVM skip
+            self.skip_zk_vm =
+                self.skip_zk_vm || self.skip_zk_vm_addresses.contains(&call.target_address);
+            if self.skip_zk_vm {
+                // reset skip_zk_vm
+                self.skip_zk_vm = false;
```

Comment:
would this reset it also if it's going to skip it anyways because it's excluded?
for example
```
skip
new Foo()
skip
foo.doSomething()
bar.wantedToBeSkipped()
```

Honeslty I don't have a problem with it, but perhaps worth documenting if for whatever reason someone tries this pattern

Reply:
This shouldn't be exposed to users, it's merely the implementation details of "skip the next call"

Reply:
Yeah I meant just document that autoskip won't make skip persist until the next call, unlike, say, prank that applies across cheatcodes. ofc prank and skipzkvm are entirely different cheatcodes, but I'm talking about "user expectations"

Reply:
But that is exactly what is in the doc for `zkVmSkip` - skips the next CREATE or CALL